### ISSUE TITLE
no-deprecated: false positive destructured array binding pattern

### ISSUE THREAD

**Issue by Mister-Hope** [oxc-project/tsgolint#889]
```ts
// Minimal reproduction: link is flagged as deprecated window.link
// But this is just a regex capture group variable name, not window.link

const DISALLOWED_CATEGORY_ITEM_REGEXP =
  /<div id="bb1"[^]+?lay-tips="选课学期:(.*?)\s*<br>\s*([^"]+?)\s*"\s+lay-iframe="(.*?)"\s+data-href="(.*?)"/g;

const content = `<div id="bb1" lay-tips="选课学期:2024春<br>一些描述" lay-iframe="iframe" data-href="/course/link"></div>`;

const result = [...content.matchAll(DISALLOWED_CATEGORY_ITEM_REGEXP)].map(
  ([, term, description, name, link]) => ({
    // link is just a variable name from regex destructuring, NOT window.link
    term,
    description,
    name,
    link, // <-- oxlint reports this as deprecated: `link` is deprecated
    canSelect: false,
  }),
);

console.log(result);
```

```
  ! typescript-eslint(no-deprecated): `link` is deprecated. A legacy feature for browser compatibility
    ,-[test.ts:10:32]
  9 | const result = [...content.matchAll(DISALLOWED_CATEGORY_ITEM_REGEXP)].map(
 10 |   ([, term, description, name, link]) => ({
    :                                ^^^^
 11 |     // link is just a variable name from regex destructuring, NOT window.link
    `----

Found 2 warnings and 0 errors.
Finished in 101ms on 1 file with 439 rules using 10 threads.
```
